core/arch/riscv64/riscv64.asm

```diff
@@ -339,16 +339,18 @@ GLOBAL_LABEL(xfer_to_new_libdr:)
         END_FUNC(xfer_to_new_libdr)
 # endif /* !STANDALONE_UNIT_TEST && !STATIC_LIBRARY */
 #endif /* UNIX */
-
+/* We need to call futex_wakeall without using any stack.
+ * Takes KSYNCH_TYPE* in a0 and the post-syscall jump target in a1
+ */
         DECLARE_FUNC(dynamorio_condvar_wake_and_jmp)
 GLOBAL_LABEL(dynamorio_condvar_wake_and_jmp:)
-        mv      REG_R9, REG_R11 /* save across syscall */
-        li      REG_R15, 0 /* arg6 */
-        li      REG_R14, 0 /* arg5 */
-        li      REG_R13, 0 /* arg4 */
-        li      REG_R12, 0x7fffffff /* arg3 = INT_MAX */
-        li      REG_R11, 1 /* arg2 = FUTEX_WAKE */
-        li      a7, 98 /* SYS_futex */
+        mv      REG_R9, ARG2 /* save across syscall */
+        li      ARG6, 0 /* arg6 */
```

Comment:
The `/* arg6 */` comments are redundant for arg 4, 5, and 6 now that we have better-named destinations.

Reply:
Done